### ISSUE TITLE
Publishing to `pub.dev` should always be done on unix-like OS

### ISSUE THREAD

**Issue by kodjodevf** [cunarist/rinf#192]
## Report

_Exception: Build process failed_

## Steps to Reproduce

dependencies:
   rinf: ^4.12.2

screenshot:
![Screenshot from 2023-10-20 18-18-18](https://github.com/cunarist/rinf/assets/107993382/e7bef70c-91a1-4bd0-8b44-9982f87aab5c)

## System Information

```
protoc --version
flutter doctor
rustc 1.73.0 (cc66ad468 2023-10-03)
libprotoc 3.12.4
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.13.8, on Ubuntu 22.04.3 LTS 6.2.0-35-generic, locale en_US.UTF-8)
[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/linux#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.

[✓] Chrome - develop for the web
[✓] Linux toolchain - develop for Linux desktop
[!] Android Studio (not installed)
[✓] VS Code (version 1.81.1)
[✓] Connected device (2 available)
[✓] Network resources
```

**Comment by temeddix** [comment]
I will inspect this issue shortly. Thanks for the report!

**Comment by temeddix** [comment]
Do you mind sharing the output from `flutter run --verbose`?

**Comment by kodjodevf** [comment]
But for some reason I don't know when I used this : 

```dependencies:
  rinf:
    git:
      url: https://github.com/cunarist/rinf.git
      ref: 4fdbc92
      path: flutter_ffi_plugin
```

It worked!!

**Comment by kodjodevf** [comment]
> Do you mind sharing the output from `flutter run --verbose`?

```
[  +35 ms] Unable to locate an Android SDK.
[  +19 ms] executing: uname -m
[   +7 ms] Exit code 0 from: uname -m
[        ] x86_64
[ +152 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[   +3 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] Artifact Instance of 'LegacyCanvasKitRemover' is not required, skipping update.
[   +4 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[ +100 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] Artifact Instance of 'LegacyCanvasKitRemover' is not required, skipping update.
[   +1 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[   +6 ms] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[ +207 ms] Skipping pub get: version match.
[  +69 ms] Found plugin rinf at /home/moustapha/.pub-cache/hosted/pub.dev/rinf-4.12.2/
[  +78 ms] Found plugin rinf at /home/moustapha/.pub-cache/hosted/pub.dev/rinf-4.12.2/
[  +66 ms] Found plugin rinf at /home/moustapha/.pub-cache/hosted/pub.dev/rinf-4.12.2/
[ +160 ms] Generating /home/moustapha/Downloads/flutter_application_1/android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java
[ +158 ms] Initializing file store
[  +15 ms] Skipping target: gen_localizations
[   +9 ms] gen_dart_plugin_registrant: Starting due to {InvalidatedReasonKind.inputChanged: The following inputs have updated contents:
/home/moustapha/Downloads/flutter_application_1/.dart_tool/package_config_subset}
[  +36 ms] Found plugin rinf at /home/moustapha/.pub-cache/hosted/pub.dev/rinf-4.12.2/
[  +21 ms] gen_dart_plugin_registrant: Complete
[   +2 ms] Skipping target: _composite
[   +2 ms] complete
[   +8 ms] Launching lib/main.dart on Linux in debug mode...
[   +6 ms] /home/moustapha/snap/flutter/common/flutter/bin/cache/dart-sdk/bin/dart --disable-dart-dev
/home/moustapha/snap/flutter/common/flutter/bin/cache/dart-sdk/bin/snapshots/frontend_server.dart.snapshot --sdk-root
/home/moustapha/snap/flutter/common/flutter/bin/cache/artifacts/engine/common/flutter_patched_sdk/ --incremental --target=flutter --experimental-emit-debug-metadata -DFLUTTER_WEB_AUTO_DETECT=true
-DFLUTTER_WEB_CANVASKIT_URL=https://www.gstatic.com/flutter-canvaskit/767d8c75e898091b925519803830fc2721658d07/ --output-dill /tmp/flutter_tools.HMOBRA/flutter_tool.HNRKDA/app.dill --packages
/home/moustapha/Downloads/flutter_application_1/.dart_tool/package_config.json -Ddart.vm.profile=false -Ddart.vm.product=false --enable-asserts --track-widget-creation --filesystem-scheme org-dartlang-root
--initialize-from-dill build/9849d3fde14f8238bc82d80453267a41.cache.dill.track.dill --verbosity=error --enable-experiment=alternative-invalidation-strategy
[  +52 ms] Building Linux application...
[  +53 ms] <- compile package:flutter_application_1/main.dart
[   +9 ms] executing: [build/linux/x64/debug/] cmake -G Ninja -DCMAKE_BUILD_TYPE=Debug -DFLUTTER_TARGET_PLATFORM=linux-x64 /home/moustapha/Downloads/flutter_application_1/linux
[ +127 ms] -- Configuring done
[  +12 ms] -- Generating done
[  +22 ms] -- Build files have been written to: /home/moustapha/Downloads/flutter_application_1/build/linux/x64/debug
[  +14 ms] executing: ninja -C build/linux/x64/debug install
[  +21 ms] ninja: Entering directory `build/linux/x64/debug'
[  +73 ms] [1/8] Generating libhub.so, _phony_
[   +4 ms] FAILED: plugins/rinf/libhub.so plugins/rinf/_phony_ /home/moustapha/Downloads/flutter_application_1/build/linux/x64/debug/plugins/rinf/libhub.so
/home/moustapha/Downloads/flutter_application_1/build/linux/x64/debug/plugins/rinf/_phony_
[  +54 ms] cd /home/moustapha/Downloads/flutter_application_1/build/linux/x64/debug/plugins/rinf && /usr/bin/cmake -E env CARGOKIT_CMAKE=/usr/bin/cmake CARGOKIT_CONFIGURATION=Debug
CARGOKIT_MANIFEST_DIR=/home/moustapha/Downloads/flutter_application_1/linux/../native/hub
CARGOKIT_TARGET_TEMP_DIR=/home/moustapha/Downloads/flutter_application_1/build/linux/x64/debug/plugins/rinf/cargokit_build
CARGOKIT_OUTPUT_DIR=/home/moustapha/Downloads/flutter_application_1/build/linux/x64/debug/plugins/rinf CARGOKIT_TARGET_PLATFORM=linux-x64
CARGOKIT_TOOL_TEMP_DIR=/home/moustapha/Downloads/flutter_application_1/build/linux/x64/debug/plugins/rinf/cargokit_build/tool CARGOKIT_ROOT_PROJECT_DIR=/home/moustapha/Downloads/flutter_application_1/linux
/home/moustapha/.pub-cache/hosted/pub.dev/rinf-4.12.2/cargokit/run_build_tool.sh build-cmake
[   +3 ms] Permission denied
[        ] [2/8] Linking C shared library plugins/rinf/shared/librinf.so
[+3463 ms] [3/8] Generating /home/moustapha/Downloads/flutter_application_1/linux/flutter/ephemeral/libflutter_linux_gtk.so,
/home/moustapha/Downloads/flutter_application_1/linux/flutter/ephemeral/flutter_linux/fl_basic_message_channel.h,
/home/moustapha/Downloads/flutter_application_1/linux/flutter/ephemeral/flutter_linux/fl_binary_codec.h,
/home/moustapha/Downloads/flutter_application_1/linux/flutter/ephemeral/flutter_linux/fl_binary_messenger.h,
/home/moustapha/Downloads/flutter_application_1/linux/flutter/ephemeral/flutter_linux/fl_dart_project.h, /home/moustapha/Downloads/flutter_application_1/linux/flutter/ephemeral/flutter_linux/fl_engine.h,
/home/moustapha/Downloads/flutter_application_1/linux/flutter/ephemeral/flutter_linux/fl_json_message_codec.h,
/home/moustapha/Downloads/flutter_application_1/linux/flutter/ephemeral/flutter_linux/fl_json_method_codec.h,
/home/moustapha/Downloads/flutter_application_1/linux/flutter/ephemeral/flutter_linux/fl_message_codec.h,
/home/moustapha/Downloads/flutter_application_1/linux/flutter/ephemeral/flutter_linux/fl_method_call.h,
/home/moustapha/Downloads/flutter_application_1/linux/flutter/ephemeral/flutter_linux/fl_method_channel.h,
/home/moustapha/Downloads/flutter_application_1/linux/flutter/ephemeral/flutter_linux/fl_method_codec.h,
/home/moustapha/Downloads/flutter_application_1/linux/flutter/ephemeral/flutter_linux/fl_method_response.h,
/home/moustapha/Downloads/flutter_application_1/linux/flutter/ephemeral/flutter_linux/fl_plugin_registrar.h,
/home/moustapha/Downloads/flutter_application_1/linux/flutter/ephemeral/flutter_linux/fl_plugin_registry.h,
/home/moustapha/Downloads/flutter_application_1/linux/flutter/ephemeral/flutter_linux/fl_standard_message_codec.h,
/home/moustapha/Downloads/flutter_application_1/linux/flutter/ephemeral/flutter_linux/fl_standard_method_codec.h,
/home/moustapha/Downloads/flutter_application_1/linux/flutter/ephemeral/flutter_linux/fl_string_codec.h, /home/moustapha/Downloads/flutter_application_1/linux/flutter/ephemeral/flutter_linux/fl_value.h,
/home/moustapha/Downloads/flutter_application_1/linux/flutter/ephemeral/flutter_linux/fl_view.h, /home/moustapha/Downloads/flutter_application_1/linux/flutter/ephemeral/flutter_linux/flutter_linux.h, _phony_
[   +3 ms] [  +63 ms] Unable to locate an Android SDK.
[        ] [  +24 ms] executing: uname -m
[        ] [  +10 ms] Exit code 0 from: uname -m
[        ] [   +1 ms] x86_64
[        ] [  +64 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] [        ] Artifact Instance of 'LegacyCanvasKitRemover' is not required, skipping update.
[        ] [   +3 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] [ +205 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[        ] [   +1 ms] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[        ] [        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] [        ] Artifact Instance of 'LegacyCanvasKitRemover' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[        ] [        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] [   +3 ms] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'PubDependencies' is not required, skipping update.
[        ] [  +79 ms] Initializing file store
[        ] [  +12 ms] Done initializing file store
[        ] [  +78 ms] Skipping target: gen_localizations
[        ] [  +17 ms] Skipping target: gen_dart_plugin_registrant
[        ] [ +810 ms] Skipping target: unpack_linux
[        ] [ +551 ms] Skipping target: kernel_snapshot
[        ] [ +388 ms] Skipping target: debug_bundle_linux-x64_assets
[        ] [   +1 ms] Persisting file store
[        ] [  +18 ms] Done persisting file store
[        ] [  +12 ms] build succeeded.
[        ] [  +16 ms] "flutter assemble" took 2,221ms.
[        ] [ +284 ms] ensureAnalyticsSent: 261ms
[        ] [        ] Running 0 shutdown hooks
[        ] [        ] Shutdown hooks complete
[        ] [   +4 ms] exiting with code 0
[   +1 ms] ninja: build stopped: subcommand failed.
[  +34 ms] Building Linux application... (completed in 3.9s)
[+10740 ms] Exception: Build process failed
[   +8 ms] "flutter run" took 15,714ms.
[   +6 ms] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
           #1      RunCommand.runCommand (package:flutter_tools/src/commands/run.dart:760:9)
           <asynchronous suspension>
           #2      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1297:27)
           <asynchronous suspension>
           #3      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #4      CommandRunner.runCommand (package:args/command_runner.dart:212:13)
           <asynchronous suspension>
           #5      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:339:9)
           <asynchronous suspension>
           #6      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #7      FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:285:5)
           <asynchronous suspension>
           #8      run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:115:9)
           <asynchronous suspension>
           #9      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #10     main (package:flutter_tools/executable.dart:90:3)
           <asynchronous suspension>
           
           
[ +185 ms] ensureAnalyticsSent: 176ms
[        ] Running 1 shutdown hook
[   +9 ms] Shutdown hooks complete
[        ] exiting with code 1
```

**Comment by temeddix** [comment]
Oh, I think it's a permission problem with files published to `pub.dev`. I've experienced that problem a few days ago, and I should probably re-clone the whole repo before publishing the new version.

I will publish a new version in 2 hours :)

**Comment by temeddix** [comment]
I can confirm the error. Working on it...

**Comment by temeddix** [comment]
Thank you for your patience :D problem is now solved in version 4.12.4 🎉

Let's keep this issue open for now, because some tests have shown that publishing to `pub.dev` on Windows messes up everything, including EOL and file permissions. Publishing to `pub.dev` needs to be done on unix-like OS, such as Linux or macOS. I will make a pull request that improves the publishing procedure soon. The new system will use Ubuntu-based (manual) GitHub actions to publish a new version.

**Comment by temeddix** [comment]
Fixed by #196 and some additional following commits
